bentoml/tensorflow.py

```diff
@@ -433,22 +436,40 @@ def _setup(self) -> None:  # type: ignore[override]
         )
         self._model = load(self.name, model_store=self._model_store)
         if not TF2:
-            self._predict_fn = self._model.signatures["serving_default"]
+            raw_predict_fn = self._model.signatures["serving_default"]
         else:
-            self._predict_fn = getattr(self._model, self._predict_fn_name)
+            raw_predict_fn = getattr(self._model, self._predict_fn_name)
+        self._predict_fn = functools.partial(raw_predict_fn, **self._partial_kwargs)
 
     # pylint: disable=arguments-differ
     def _run_batch(  # type: ignore[override]
-        self, input_data: t.Union[t.List[t.Union[int, float]], np.ndarray, tf.Tensor]
+        self,
+        *args: t.Union[t.List[t.Union[int, float]], np.ndarray, tf.Tensor],
+        **kwargs: t.Union[t.List[t.Union[int, float]], np.ndarray, tf.Tensor],
     ) -> np.ndarray:
-        with self._device:
-            if not isinstance(input_data, tf.Tensor):
-                input_data = tf.convert_to_tensor(input_data, dtype=tf.float32)
+
+        params = Params[t.Union[t.List[t.Union[int, float]], np.ndarray, tf.Tensor]](
+            *args, **kwargs
+        )
+
+        with tf.device(self._device_id):
+
+            def _mapping(item) -> tf.Tensor:
```

Comment:
```suggestion
            def _mapping(item: t.Union[t.List[t.Union[int, float]], "np.ndarray[t.Any, np.dtype[t.Any]]", tf.Tensor) -> tf.Tensor:
```

Reply:
can we just use `np.ndarray`?

Reply:
mainly for numpy > 1.20, thats how you define numpy types. the other stubs PR i'm working on atm will basically have newer stubs for backward compat. 
Pyright will throw an error where type of `np.ndarray` would be unknown